src/NUnitFramework/tests/Assertions/AssertFailTests.cs

```diff
@@ -104,13 +104,13 @@ public void AssertFailWorksWithMessageAndArgs()
             Assert.AreEqual(ResultState.Failure, result.ResultState);
             Assert.AreEqual("MESSAGE: 2+2=4", result.Message);
 
-            Assert.AreEqual(result.AssertionResults.Count, 1);
+            Assert.AreEqual(1, result.AssertionResults.Count);
             var assertion = result.AssertionResults[0];
             Assert.That(assertion.Status, Is.EqualTo(AssertionStatus.Failed));
             Assert.That(assertion.Message, Is.EqualTo("MESSAGE: 2+2=4"));
         }
 
-        [Test]
+        [Test, Ignore("Currently Fails: Must use Assert.Catch or Assert.Throws")]
```

Comment:
@CharliePoole Since we've been telling folks not to rely on this behavior, shouldn't we switch this from TODO to by-design (remove Ignore and negate the test)?

Reply:
Fixed by https://github.com/nunit/nunit/pull/2924.